src/workflows-app/components/ImportGithub.ts

```diff
@@ -24,18 +25,29 @@ const constraints = {
   },
 };
 
-const ImportGithub = ({ setLoading, signal, workspace, name, namespace, setSelectedSubHeader }) => {
-  const [methodName, setMethodName] = useState('');
-  const [methodVersionName, setMethodVersionName] = useState('');
-  const [methodUrl, setMethodUrl] = useState('');
-  const [methodUrlModified, setMethodUrlModified] = useState(false);
-  const [methodNameModified, setMethodNameModified] = useState(false);
-  const [methodId, setMethodId] = useState('');
+export interface ImportGithubProps {
+  workspace: WorkspaceWrapper;
+  namespace: string;
+  setSelectedSubHeader: ImportWorkflowModalProps['setSelectedSubHeader'];
+  setLoading: (value: boolean) => void;
+  signal?: AbortSignal;
+  name: string;
+}
 
-  const [importWorkflowModal, setImportWorkflowModal] = useState(false);
-  const [importLoading, setImportLoading] = useState(false);
-  const [successfulImport, setSuccessfulImport] = useState(undefined);
-  const [errorMessage, setErrorMessage] = useState('');
+const ImportGithub: React.FC<ImportGithubProps> = (props): React.ReactNode => {
+  const { setLoading, signal, workspace, name, namespace, setSelectedSubHeader } = props;
```

Comment:
shifted this and other component signatures to use recommended type signature pattern, and avoid decomposing props inline.  Instead props decomposition is done as the 1st line inside the component function body to improve readability.